openff/toolkit/topology/molecule.py

```diff
@@ -445,18 +445,25 @@ def is_bonded_to(self, atom2):
                     return True
         return False
 
-    @property
-    def is_in_ring(self):
+    def is_in_ring(self, toolkit_registry=GLOBAL_TOOLKIT_REGISTRY) -> bool:
         """
         Return whether or not this atom is in a ring(s) (of any size)
 
+        Currently only implemented via The RDKit, which must be installed.
+
         """
         if self._molecule is None:
             raise NotAttachedToMoleculeError(
                 "This Atom does not belong to a Molecule object"
             )
 
-        return any([self.molecule_atom_index in ring for ring in self._molecule.rings])
+        _is_in_ring = toolkit_registry.call(
+            "atom_is_in_ring",
+            self._molecule,
+            self.molecule_atom_index,
```

Comment:
Here and for `bond.is_in_ring`, we can just pass the `Atom` or `Bond`, since those know their `Molecule`

```suggestion
            self
```